src/wallet/wallet.cpp

```diff
@@ -4126,7 +4126,9 @@ bool CWallet::UpgradeWallet(int version, bilingual_str& error, std::vector<bilin
     }
 
     // Permanently upgrade to the version
-    SetMinVersion(GetClosestWalletFeature(version));
+    const WalletFeature new_version{GetClosestWalletFeature(version)};
+    SetMinVersion(new_version);
+    version = static_cast<int>(new_version); // return actual updated version
```

Comment:
Is `static_cast` necessary here? `WalletFeature` should already be an int.

Reply:
Oh you're right, looks like we can just do `version = new_version` here. Will test a push without the cast. Thanks!